nvi-rest/src/main/java/no/sikt/nva/nvi/rest/fetch/FetchReportStatusByPublicationIdHandler.java

```diff
@@ -0,0 +1,77 @@
+package no.sikt.nva.nvi.rest.fetch;
+
+import static java.net.HttpURLConnection.HTTP_OK;
+import static no.sikt.nva.nvi.common.db.DynamoRepository.defaultDynamoClient;
+import static nva.commons.core.attempt.Try.attempt;
+import com.amazonaws.services.lambda.runtime.Context;
+import java.net.URI;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import no.sikt.nva.nvi.common.db.CandidateRepository;
+import no.sikt.nva.nvi.common.db.PeriodRepository;
+import no.sikt.nva.nvi.common.service.exception.CandidateNotFoundException;
+import no.sikt.nva.nvi.common.service.model.Candidate;
+import no.sikt.nva.nvi.common.utils.ExceptionMapper;
+import no.sikt.nva.nvi.rest.fetch.ReportStatusDto.StatusDto;
+import nva.commons.apigateway.ApiGatewayHandler;
+import nva.commons.apigateway.RequestInfo;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.attempt.Failure;
+
+public class FetchReportStatusByPublicationIdHandler extends ApiGatewayHandler<Void, ReportStatusDto> {
+
+    private static final String PATH_PARAM_PUBLICATION_ID = "publicationId";
+    private final CandidateRepository candidateRepository;
+    private final PeriodRepository periodRepository;
+
+    @JacocoGenerated
+    public FetchReportStatusByPublicationIdHandler() {
+        this(new CandidateRepository(defaultDynamoClient()), new PeriodRepository(defaultDynamoClient()));
+    }
+
+    public FetchReportStatusByPublicationIdHandler(CandidateRepository candidateRepository,
+                                                   PeriodRepository periodRepository) {
+        super(Void.class);
+        this.candidateRepository = candidateRepository;
+        this.periodRepository = periodRepository;
+    }
+
+    @Override
+    protected void validateRequest(Void unused, RequestInfo requestInfo, Context context) throws ApiGatewayException {
+
+    }
+
+    @Override
+    protected ReportStatusDto processInput(Void unused, RequestInfo requestInfo, Context context)
+        throws ApiGatewayException {
+        var publicationId = getPublicationId(requestInfo);
+        return attempt(() -> publicationId)
```

Comment:
I think you can mode second attempt line to the first one, cause first one does not throw exception? Or getPublicationId() inside of attempt

Reply:
Ups, yes. Refactoring tail:)